Grbl_Esp32/src/MotionControl.cpp

```diff
@@ -313,59 +313,24 @@ void mc_homing_cycle(uint8_t cycle_mask) {
     else
 #endif
     {
-        // Search to engage all axes limit switches at faster homing seek rate.
-        if (!axis_is_squared(HOMING_CYCLE_0)) {
-            limits_go_home(HOMING_CYCLE_0);  // Homing cycle 0
-        } else {
-            ganged_mode           = SquaringMode::Dual;
-            n_homing_locate_cycle = 0;  // don't do a second touch cycle
-            limits_go_home(HOMING_CYCLE_0);
-            ganged_mode           = SquaringMode::A;
-            n_homing_locate_cycle = NHomingLocateCycle;  // restore to default value
-            limits_go_home(HOMING_CYCLE_0);
-            ganged_mode = SquaringMode::B;
-            limits_go_home(HOMING_CYCLE_0);
-            ganged_mode = SquaringMode::Dual;  // always return to dual
-        }
-#ifdef HOMING_CYCLE_1
-        if (!axis_is_squared(HOMING_CYCLE_1)) {
-            limits_go_home(HOMING_CYCLE_1);
-        } else {
-            ganged_mode           = SquaringMode::Dual;
-            n_homing_locate_cycle = 0;  // don't do a second touch cycle
-            limits_go_home(HOMING_CYCLE_1);
-            ganged_mode           = SquaringMode::A;
-            n_homing_locate_cycle = NHomingLocateCycle;  // restore to default value
-            limits_go_home(HOMING_CYCLE_1);
-            ganged_mode = SquaringMode::B;
-            limits_go_home(HOMING_CYCLE_1);
-            ganged_mode = SquaringMode::Dual;  // always return to dual
-        }
-#endif
-#ifdef HOMING_CYCLE_2
-        if (!axis_is_squared(HOMING_CYCLE_2)) {
-            limits_go_home(HOMING_CYCLE_2);
-        } else {
-            ganged_mode           = SquaringMode::Dual;
-            n_homing_locate_cycle = 0;  // don't do a second touch cycle
-            limits_go_home(HOMING_CYCLE_2);
-            ganged_mode           = SquaringMode::A;
-            n_homing_locate_cycle = NHomingLocateCycle;  // restore to default value
-            limits_go_home(HOMING_CYCLE_2);
-            ganged_mode = SquaringMode::B;
-            limits_go_home(HOMING_CYCLE_2);
-            ganged_mode = SquaringMode::Dual;  // always return to dual
+        for (uint8_t cycle = 0; cycle < MAX_N_AXIS; cycle++) {
+            uint8_t homing_mask = homing_cycle[cycle]->get();
```

Comment:
This could be "auto homing_mask =" .  That way it would not have to change if we should ever decide to change the size of masks.  I sort of have this long-term plan to get rid of most of the uint8_t's that are sprinkled all over the place.

Reply:
Will do